crates/core/tedge/src/cli/mqtt/error.rs

```diff
@@ -14,4 +16,57 @@ pub enum MqttError {
 
     #[error("MQTT connection error: {0}\n\nHint: Is MQTT server running?")]
     ServerConnection(String),
+
+    #[error("Failed to create a TLS config")]
+    TlsConfig(#[from] CertificateError),
+
+    #[error("Server TLS certificate could not be verified")]
+    ServerCertVerify(#[from] ServerCertVerifyError),
+}
+
+impl From<rumqttc::TlsError> for MqttError {
+    fn from(error: rumqttc::TlsError) -> Self {
+        Self::ServerCertVerify(ServerCertVerifyError::from(error))
+    }
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum ServerCertVerifyError {
+    /// Received certificate is not in X509v3 format
+    #[error("Received certificate is not in X.509 v3 format")]
+    UnsupportedCertVersion,
+
+    /// Received certificate is not valid for the name it is being validated for.
+    #[error(
+        "Received certificate is not valid for the name it is being validated for.
+        \nHint: Be sure to use a hostname specified in the CN field of the certificate and for the certificate to have a Subject Alt Name section."
+    )]
```

Comment:
This error message is written from the point of view of the guy who created the broker certificate. It might be less useful for an end-user who used the wrong hostname to connect the broker.

```suggestion
    #[error(
        "The MQTT server certificate is not valid as not matching the server hostname.
        \nHint: Be sure to use a hostname specified in the CN field of the certificate or one the Subject Alt Name section."
    )]
```

Reply:
Thanks for the clarification.

Reply:
Error message was reworded according to the recommendations and error structure was changed a bit.